### PR TITLE
configure random seed dtype based on backend

### PR DESCRIPTION
**Pull request by haohuanw**
seeing below error on running tensorflow distributed training with multiple workers. the issue being seed state getting broadcasted is in a dtype that tensorflow doesn't support:

```
Exception encountered: ''Value for attr 'T' of uint32 is not in the list of allowed values: bool, float, half, double, int32, int64
	; NodeDef: {{node CollectiveBcastSend}}; Op<name=CollectiveBcastSend; signature=input:T -> data:T; attr=T:type,allowed=[DT_BOOL, DT_FLOAT, DT_HALF, DT_DOUBLE, DT_INT32, DT_INT64]; attr=group_size:int; attr=group_key:int; attr=instance_key:int; attr=shape:shape; attr=communication_hint:string,default="auto"; attr=timeout_seconds:float,default=0; is_stateful=true; is_distributed_communication=true> [Op:CollectiveBcastSend]''
```

this pr introduces a seed dtype function to customize for an ideal seed dtype for backends.

### CODE REVIEW

File: keras/src/backend/tensorflow/random.py [hunk 1]
```diff
@@ -14,15 +14,15 @@ def tf_draw_seed(seed):
 
 def normal(shape, mean=0.0, stddev=1.0, dtype=None, seed=None):
     dtype = dtype or floatx()
-    seed = tf_draw_seed(seed)
```
Comment (fchollet) [inline keras/src/backend/tensorflow/random.py:17]: You can remove the tf_draw_seed function then

File: keras/src/backend/jax/core.py [hunk 2]
```diff
@@ -346,6 +346,11 @@ def unstack(x, num=None, axis=0):
     ]
 
 
+def random_seed_dtype():
+    # jax random seed uses uint32.
+    return standardize_dtype("uint32")
```
Comment (fchollet) [inline keras/src/backend/jax/core.py:351]: `standardize_dtype` will be a no-op here. You can skip it.

**Review by fchollet — COMMENTED** [review]
Thanks for the PR. This is a reasonable change.

**Review by fchollet — APPROVED** [review]
LGTM, thank you

**Comment by fchollet** [comment]
It seems this breaks TF GPU CI due to a strange issue -- TF automatically places `int32` constants on CPU (this is different compared to every other dtype). I think using `int64` instead would work -- trying it now.
